pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -158,16 +159,39 @@ func getSubComponentTypeByImageName(imageName string) string {
 	return retValue
 }
 
-func setObjectLabels(cr *api.KieApp, object *api.CustomObject) {
+func setObjectLabels(cr *api.KieApp, object *api.CustomObject, subcomponent string) {
 	for index, obj := range object.DeploymentConfigs {
-		subcomponent, _, _ := GetImage(object.DeploymentConfigs[index].Spec.Template.Spec.Containers[0].Image)
 		object.DeploymentConfigs[index].Spec.Template.Labels = setLabels(cr, obj.Spec.Template.Labels, subcomponent, getSubComponentTypeByImageName(subcomponent))
 	}
 	for index, obj := range object.StatefulSets {
-		subcomponent, _, _ := GetImage(object.StatefulSets[index].Spec.Template.Spec.Containers[0].Image)
 		object.StatefulSets[index].Spec.Template.Labels = setLabels(cr, obj.Spec.Template.Labels, subcomponent, getSubComponentTypeByImageName(subcomponent))
 	}
+}
+
+func setObjectLabelsForServer(cr *api.KieApp, object *api.CustomObject) {
+	for index, obj := range object.DeploymentConfigs {
+		subcomponent := getFormattedComponentName(cr, object.DeploymentConfigs[index].Name)
```

Comment:
this won't work, as a user can name their kieservers however they like. a user could name their kieserver `foo` if they like. instead, something like this might work?
```golang
		subcomponent := getFormattedComponentName(cr, "kieserver")
```